product_status/models/product_state.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class ProductState(models.Model):
+    _inherit = "product.state"
+
+    def write(self, vals):
+        allow = {"default", "active"}
+        if not allow.issuperset(set(vals)):
+            self._check_module_data()
+        return super().write(vals)
+
+    def unlink(self):
+        self._check_module_data()
+        return super().unlink()
+
+    def _check_module_data(self):
+        if self.env.user._is_superuser():
+            return True
+        default_data = [st.code for st in self._get_module_data()]
```

Comment:
```suggestion
        module_data = self._get_module_data()
        module_data.fetch(['name', 'code'])
        default_data = [st.code for st in module_data]
```
in old code, although we only want value from column `code`, but odoo will get all column in table `product_state`
![image](https://github.com/user-attachments/assets/1b1f4821-52d2-4a1d-bed0-e46328bc65e3)

in new code, we only select value from column `code` by method `fetch`, we will optimize code
![image](https://github.com/user-attachments/assets/73f5b075-9a1a-4e63-88f9-2297dc38f6fd)